### PR TITLE
Feature/76 youtube

### LINKED ISSUE #76: Notify on new YouTube video
**Is your feature request related to a problem? Please describe.**
n/a

**Describe the solution you'd like**
Add a new notification list for each YouTube channel you wish to be notified of. When a new YouTube video is posted, the appropriate notification list is called.

This is done by using the YouTube API to get the latest video's that are uploaded to the appropriate channel. Then the video ID is checked against the saved video ID to see if it changed.

**Describe alternatives you've considered**
Do it all manually.

**Additional context**
This would mostly be used to notify for a new Koterij video by Henk Rijckaert.

### PR DESCRIPTION
**Pull request by DrSkunk**
Adds youtube notification capabilities.

Adds the following commands
- `add_youtube <YouTube channel ID> <text channel ID or name>`
- `remove_youtube <YouTube channel ID>`
- `list_youtube`

### CODE REVIEW

File: Commands/youtube.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    @commands.command(
+        name="remove_youtube",
+        usage="remove_youtube_usage",
+        help="remove_youtube_help",
+    )
+    async def remove_youtube_channel(
+        self, ctx: commands.Context, youtube_channel_id: str, text_channel: str
+    ):
+        """
+        Add a Youtube channel to be notified
```
Comment (bobvdvleuten) [inline Commands/youtube.py]: nitpicking here: doc says add when function is remove ;)

File: Commands/youtube.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        guild_data = await get_guild_data(ctx.message.guild.id)
+        # Error if not admin
+        if not guild_data.user_is_admin(ctx.author):
+            gif = translate("not_admin_gif", await culture(ctx))
+            return await ctx.send(gif)
+
+        remove_response = await guild_data.remove_youtube_channel(youtube_channel_id)
+        msg = ""
+        if remove_response:
+            msg = f"Removed notifier for `{youtube_channel_id}`"
```
Comment (bobvdvleuten) [inline Commands/youtube.py]: Ideally should be translated as well, no?

File: Commands/youtube.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        else:
+            msg = f"There is no notifier for `{youtube_channel_id}`"
```
Comment (bobvdvleuten) [inline Commands/youtube.py]: same comment as 89

File: Commands/youtube.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        List all Youtube channels that are being monitored
+        """
+
+        guild_data = await get_guild_data(ctx.message.guild.id)
+        msg = translate("youtube_list_title", await culture(ctx))
+
+        for channel_id, channel_data in guild_data.youtube_channels.items():
+            msg = (
+                msg
+                + f"\n - Channel `channel_id` posts in <#{channel_data['text_channel_id']}>, last video ID: `{channel_data['latest_video_id']}`"
```
Comment (bobvdvleuten) [inline Commands/youtube.py:110]: Same comment as 89

File: Translations/Translations.csv [hunk 5]
```diff
@@ -93,4 +93,14 @@
...
 "notif_too_long","Your message is {0} characters too long for the Discord bot","Je bericht is {0} tekens te lang, probeer met een korter bericht."
+"add_youtube_usage","<YouTube channel ID> <text channel ID or name>","<YouTubekanaal ID> <tekstkanaal ID of naam>"
+"add_youtube_help","**admin-only** \n \u2003 Add a new YouTube notification listener. Every 10 minutes there is a check for new posted videos.","**admin-functie** \n \u2003 Voeg een nieuwe YouTube melder voor nieuwe video's toe. Elke 10 minuten wordt er gekeken of er nieuwe video's zijn."
+"remove_youtube_usage","<YouTube channel ID>","<YouTubekanaal ID>"
+"remove_youtube_help","**admin-only** \n \u2003 Remove a YouTube notification listener","**admin-functie** \n \u2003 Verwijder een YouTube melder voor nieuwe video's"
+"list_youtube_help","Give an overview of all new YouTube video notifiers.","Geef een overzicht van alle YouTube nieuwe video melders"
+"youtube_list_title","The following YouTube notifiers are currently active:","De volgende YouTube melders zijn momenteel actief:"
+"youtube_added","Added a notifier for `{0}` to be posted in text channel {1}","Notificaties werden toegevoegd voor `{0}` in het #{1} kanaal"
```
Comment (ThunderbirdOne) [inline Translations/Translations.csv]: Spoorwegteken voor het engelse kanaal? Ofwel in NL weghalen :D

**Review by ThunderbirdOne — CHANGES_REQUESTED** [review]
Looks good aside from some textual changes.

When completed, don't forget to complete YOUTUBE TOKEN in .env file!

**Review by ThunderbirdOne — APPROVED** [review]
🚢

File: Commands/youtube.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        latest_video = await get_latest_video(youtube_channel_id)
+
+        # Retrieve channel
+        channel = ctx.bot.get_channel(int(text_channel))
+        if not channel:
+            channel = discord.utils.get(ctx.channel.guild.channels, name=text_channel)
+        
+        # TODO: Give information to the user when the text channel does not exist
+        if not channel:
+            raise Exception("Invalid text channel provided")
```
Comment (bobvdvleuten) [inline Commands/youtube.py:51]: This will leave a "da kennek nie foump" generic error message in Discord, no, if only handled by an Exception?

**Review by bobvdvleuten — APPROVED** [review]
Had one minor comment but since there's a TODO with it and the functionality is admin only, i think we can go ahead with this.

**Pull request merged**